mqtt/mqtt-bridge/src/persist/waking_state/ring_buffer/mod.rs

```diff
@@ -415,8 +415,8 @@ fn find_pointers_and_order_post_crash(file: &mut File, max_file_size: u64) -> Ri
     let block_size = *SERIALIZED_BLOCK_SIZE;
     let mut start = inner.write_index();
     let mut end = start + block_size;
-    let mut read = 0;
-    let mut write = 0;
+    let mut read = start;
+    let mut write = start;
```

Comment:
Can you give some context as to why this is changing?

Reply:
Yeah, we already found the first block at this point, which has a write_index (this is the position of the write ptr)
So either:
A) the write ptr is at 0
B) the write ptr is at an offset away from 0

To elaborate further, we know that the read and write ptr must start at the beginning of a block. We have found the first block in the RB as we had scanned from 0->X to find it. If X is greater than 0, then it means all the bytes before X are likely data or an incomplete block, therefore we know starting at 0 is useless. That is why these now start at `start` instead of `0`, so that these begin at the first block we found and not potentially in middle of garbled bytes that mean nothing.

Sorry if this is still a bit complicated... Really looking forward to when I replace all of this with a simpler design :)

Reply:
> We have found the first block in the RB as we had scanned from 0->X to find it. If X is greater than 0, then it means all the bytes before X are likely data or an incomplete block

If there is data before X, then shouldn't the read pointer be positioned before the write pointer so that we can read this data? I was expecting this to re-establish these pointers at specific locations, not necessarily the beginning of the block.

Reply:
Ptrs have to start at a block is what I mean, perhaps I should've worded it as "... bytes before X are not a block"